pvnet/models/base_model.py

```diff
@@ -609,12 +613,48 @@ def _log_forecast_plot(self, batch, y_hat, accum_batch_num, timesteps_to_plot, p
             print(e)
         plt.close(fig)
 
+    def _log_validation_results(self, batch, y_hat, accum_batch_num):
+        """Append validation results to self.validation_epoch_results"""
+
+        y = batch[self._target_key][:, -self.forecast_len :, 0]
+        batch_size = y.shape[0]
+
+        for i in range(batch_size):
+            y_i = y[i].detach().cpu().numpy()
+            y_hat_i = y_hat[i].detach().cpu().numpy()
+
+            time_utc_key = BatchKey[f"{self._target_key_name}_time_utc"]
+            time_utc = batch[time_utc_key][i, -self.forecast_len :].detach().cpu().numpy()
+
+            id_key = BatchKey[f"{self._target_key_name}_id"]
+            target_id = batch[id_key][i].detach().cpu().numpy()
+            if target_id.ndim > 0:
+                target_id = target_id[0]
+
+            results_df = pd.DataFrame(
+                {
+                    "y": y_i,
+                    "y_hat": y_hat_i,
```

Comment:
Have you done a training run with this? When predicting quantiles, I think `y_i` and `y_hat_i` will be different shapes

Reply:
I havent done training, but the end2end test does go through here. 

becasue i take `y = batch[self._target_key][:, -self.forecast_len :, 0]`, it makes it the same length as y_hat. 

Perhaps theres a better way to standardised that. something for the future

Reply:
Yeh but I think in this case, `y_i` is a vector with shape `(horizon_step,)`. But `y_hat_i` can either be a vector with shape `(horizon_step,)` or `(horizon_step, quantile,)` depending on whether we are training to predict quantiles or only a central value. The end2end test only tests non-quantile training.

Reply:
oh good point. ill have a think about the qunalite things. Good catch on that

Reply:
I should have it now